src/modules/base/ResourcesIndex.ts

```diff
@@ -0,0 +1,232 @@
+ * applies to Drash version 3.X.X and any later version.
+ *
+ * This file is part of Drash. See <https://github.com/drashland/drash>.
+ *
+ * Drash is free software: you can redistribute it and/or modify it under the
+ * terms of the GNU General Public License as published by the Free Software
+ * Foundation, either version 3 of the License, or (at your option) any later
+ * version.
+ *
+ * Drash is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+ * A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * Drash. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+// Imports > Core
+import { IResource } from "../../core/Interfaces.ts";
+import { ConstructorWithArgs } from "../../core/Types.ts";
+
+// Imports > Standard
+import { ConsoleLogger, Level } from "../../standard/log/ConsoleLogger.ts";
+import { HTTPError } from "../../standard/errors/HTTPError.ts";
+import { Logger } from "../../standard/log/Logger.ts";
+import { SearchIndex } from "../../standard/handlers/SearchIndex.ts";
+import { StatusCode } from "../../standard/http/response/StatusCode.ts";
+
+// Imports > Modules
+import { ResourceClassesArray } from "./types/ResourceClassesArray.ts";
+
+type Index = {
+  resource: IResource;
+  path_patterns: PathPattern[];
+};
+
+type Input = { url: string };
+
+type PathPatternExecResult = {
+  pathname?: {
+    groups: Record<string, string | undefined>;
+  };
+};
+
+interface PathPattern {
+  pathname: string;
+  exec(input: string): PathPatternExecResult | null;
+}
+
+type PathPatternClass = ConstructorWithArgs<
+  PathPattern,
+  [{ pathname: string }]
+>;
+
+type SearchResult = {
+  resource: IResource;
+  path_params: Record<string, string | undefined>;
+};
+
+abstract class ResourcesIndex extends SearchIndex<
+  Promise<SearchResult | null>
+> {
+  #cached_search_results: Record<string, SearchResult | null> = {};
+  #logger: Logger = ConsoleLogger.create("ResourcesIndex", Level.Off);
+  protected index: Index[] = [];
+  protected resources: ResourceClassesArray = [];
+  protected PathPatternClass: PathPatternClass;
+
+  constructor(
+    PathPatternClass: PathPatternClass,
+    resources: ResourceClassesArray,
+  ) {
+    super();
+    this.resources = resources ?? [];
+    this.PathPatternClass = PathPatternClass;
+    this.buildIndex(this.resources);
+  }
+
+  public handle(request: Input): Promise<SearchResult | null> {
+    return Promise
+      .resolve()
+      .then(() => this.#logger.debug(`Input received`))
+      .then(() => this.#validateRequest(request))
+      .then(() => this.search(request))
+      .then((result) => super.nextHandler({ request, result }));
+  }
+
+  protected buildIndex(resources: ResourceClassesArray): void {
+    this.#logger.debug(`Building resources index`);
```

Comment:
Another instance of logging that should not be the default behaviour.

Reply:
done in https://github.com/crookse/fork-drash/pull/1/commits/60e2c967262636b6b409618c3cd5b051e01312db and https://github.com/drashland/drash/pull/698/commits/daea29f90c5f3a8d415c8be08df050612c14b7af (same commits as above)